src/classes/asettings.h

```diff
@@ -24,20 +24,180 @@
  * \brief Thin (namespace) wrapper for the QSettings class,
  * simplifying reading and writing of settings.
  */
-namespace ASettings
-{
+class ASettings {
+public:
+    enum class Setup {
+        SetupComplete,
+    };
 
-/*!
- * \brief Should be called after QCoreApplication::set...Name have been called.
- */
-void setup();
+    enum class Main {
+        Theme,
+        ThemeID,
+    };
+
+    enum class LogBook {
+        View,
+    };
+
+    enum class UserData {
+        LastName,
+        FirstName,
+        Company,
+        EmployeeID,
+        Phone,
+        Email,
+        DisplaySelfAs,
+        Alias,
+        AcSortColumn, // not sure what ac is
+        PilSortColumn,
+        AcAllowIncomplete,
+    };
+
+    enum class FlightLogging {
+        Function,
+        Approach,
+        NightLogging,
+        LogIFR,
+        FlightNumberPrefix,
+        NumberTakeoffs,
+        NumberLandings,
+        PopupCalendar,
+        PilotFlying,
+        NightAngle,
+        Rules,
+    };
+
+    enum class NewFlight {
+        FunctionComboBox,
+        CalendarCheckBox,
+    };
+
+    /*!
+     * \brief Should be called after QCoreApplication::set...Name have been called.
+     */
+    static void setup();
+
+    static QVariant read(const Setup key);
+    static void write(const Setup key, const QVariant &val);
+
+    static QVariant read(const LogBook key);
+    static void write(const LogBook key, const QVariant &val);
+
+    static QVariant read(const Main key);
+    static void write(const Main key, const QVariant &val);
+
+    static QVariant read(const FlightLogging key);
+    static void write(const UserData key, const QVariant &val);
+
+    static QVariant read(const UserData key);
+    static void write(const FlightLogging key, const QVariant &val);
+
+    static QVariant read(const NewFlight key);
+    static void write(const NewFlight key, const QVariant &val);
+
+    // [G]: enum class may be making it abit too strict perhaps?
```

Comment:
I prefer the enum class approach as you laid out above. It is more accessible (human readable) and easier to interact with. Using a string-based approach is also closer to the settings file itself. After all, it's just a plain text file and since the program is still quite volatile as you correctly pointed out, I think it is better to work with a more accessible way of accessing the settings - Keep it simple stupid and let's consider revisiting this topic at a later stage!